### PR TITLE
Allow for property tables to live on a separate db

### PR DESCRIPTION
**Pull request by HeyNonster**
This PR adds the ability for the property tables of a model to live on a separate database to the model itself.

It does so by adding `attr_accessor` (to the class level) of
`ActiveRecord::Base` that allows you to set a specific connection class
for the property sets tables.

This is useful if you want to have the tables for your model live on one
database, but all of the property sets tables live on a separate database.

Usage:

```ruby
class Account < MainConnectionClass
  # Ensure you set this _before_ configuring the property sets.
  self.property_sets_connection_class = SeparateDatabase

  property_set :settings do
    property :foo
  end
end
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -154,6 +154,35 @@ end
...
+end
+
+class SeparateDatabase < ActiveRecord::Base
+  self.abstract_class = true
+
+  connects_to(database: { writing: bar })
+end
+
+class Account < MainConnectionClass
+  # Ensure you set this _before_ configuring the property sets.
```
Comment (Thomascountz) [inline README.md:175]: 👍🏾

File: lib/property_sets/active_record_extension.rb [hunk 2]
```diff
@@ -5,6 +5,8 @@
 module PropertySets
   module ActiveRecordExtension
     module ClassMethods
+      attr_accessor :property_sets_connection_class
```
Comment (Thomascountz) [inline lib/property_sets/active_record_extension.rb:8]: I don't remember where we landed on throwing a warning if `property_sets_connection_class=` gets called more than once?

```ruby
      def property_sets_connection_class=(value)
        warn "property_sets_connection_class has already been set!" if @property_sets_connection_class
        @property_sets_connection_class = value
      end
      ```

Comment (HeyNonster) [reply lib/property_sets/active_record_extension.rb:8]: It seems a little too complicated to set up and make work with Rails 6.0, tbh.

Comment (Thomascountz) [reply lib/property_sets/active_record_extension.rb:8]: Ah, right. Now I remember there was some issue specific to 6.0. I don't remember what it was through 😅. But, I agree that this small nicety isn't worth the additional effort 😄

**Review by Thomascountz — APPROVED** [review]
🚀 Small nit. Shit it!

File: spec/property_sets_spec.rb [hunk 3]
```diff
@@ -111,7 +116,11 @@ class AnotherThing < ActiveRecord::Base
   end
 
   it "reject settings with an invalid name" do
-    s = Parent::AccountSetting.new(:account => account)
+    # Because we are running these specs with two separate classes
+    # (Parent::Account & Parent::AccountAltDb), we need to build the
+    # settings class class name manually.
+    settings_klass = Object.const_get("#{account_klass.to_s}Setting")
```
Comment (bquorning) [inline spec/property_sets_spec.rb:122]:
```suggestion
    settings_klass = Object.const_get("#{account_klass}Setting")
```

**Review by bquorning — APPROVED** [review]

**Pull request merged**